packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts

```diff
@@ -0,0 +1,126 @@
+  .requiredOption("-m, --merkleDistributorAddress <address>", "address of the merkle distributor contract")
+  .parse(process.argv);
+
+const options = program.opts();
+
+const claimsObject = JSON.parse(fs.readFileSync(options.input, { encoding: "utf8" }));
+
+// We can't easily do runtime verification of JSON file types in typescript. We could use a JSON scheme, but to keep
+// things simple for now we can just double check that some important keys are present within the JSON file.
+if (typeof claimsObject !== "object") throw new Error("Invalid JSON");
+const expectedKeys = ["chainId", "rewardToken", "windowIndex", "totalRewardsDistributed", "windowStart", "claims"];
+expectedKeys.forEach(expectedKey => {
+  if (!Object.keys(claimsObject).includes(expectedKey)) {
+    throw new Error("claims object missing expected key");
+  }
+});
+
+const merkleDistributor = new web3.eth.Contract(getAbi("MerkleDistributor"), options.merkleDistributorAddress);
+
+async function main() {
+  const account = (await web3.eth.getAccounts())[0];
+
+  // Do some basic sanity checks.
+  console.log("0. Running some basic sanity checks on the payout file 🔎");
+  assert(web3.utils.isAddress(options.merkleDistributorAddress), "Invalid merkleDistributorAddress");
+  assert((await merkleDistributor.methods.owner().call()) == account, "Unlocked account does not own the distributor");
+  assert((await web3.eth.net.getId()) == claimsObject.chainId, "The connected network not match your  JSON chainId");
+  assert(web3.utils.isAddress(claimsObject.rewardToken), "Invalid rewardToken" + claimsObject.rewardToken);
+  assert((await merkleDistributor.methods.lastSeededIndex().call()) == claimsObject.windowIndex, "Wrong windowIndex");
+
+  console.log("File passed the checks\n\n1. Adding claims file to IPFS 🛰");
+
+  const ipfsHash = await IpfsHelper.uploadFile(JSON.stringify(claimsObject.claims));
+
+  console.log("Claims file added to IPFS with hash:", ipfsHash, "\n\n2. Pinning claims file to IPFS 📌");
+
+  await IpfsHelper.pinHash(ipfsHash);
+
+  console.log("Claims file pinned to IPFS!\n\n3. Adding claims file to cloudflare KV 🗺");
+
+  await CloudflareKVHelper.addClaimsToKV(claimsObject.claims, claimsObject.chainId, claimsObject.windowIndex);
+
+  console.log("Claims file added to cloudflareKV!\n\n4. Updating the lookup claims indices 🔭");
+
+  await CloudflareKVHelper.updateChainWindowIndicesFromKV(
+    claimsObject.chainId,
+    claimsObject.windowIndex,
+    ipfsHash,
+    claimsObject.rewardToken,
+    claimsObject.totalRewardsDistributed,
+    claimsObject.windowStart
+  );
+
+  console.log("5. Checking allowance in payment token 💸");
+  const rewardToken = new web3.eth.Contract(getAbi("ExpandedERC20"), claimsObject.rewardToken);
+  const currentRewardAllowance = await rewardToken.methods.allowance(account, options.merkleDistributorAddress).call();
+
+  if (toBN(currentRewardAllowance).lt(toBN(MAX_UINT_VAL).div(toBN("2")))) {
```

Comment:
Maybe just compare `currentRewardAllowance` to `claimsObject.totalRewardsDistributed`

Reply:
This would mean needing to re-approve every time. I opted for this method to ensure that we don't have to send multiple approval transactions. I am happy to do it that way if you think it better.